python/desc/imsim/camera_readout.py

```diff
@@ -206,43 +237,47 @@ def _make_amp_image(self, amp_name):
         amp_name : str
             The amplifier name, e.g., "R22_S11_C00".
         """
-        amp_props = self.fp_props.get_amp(amp_name)
-        bbox = amp_props.mosaic_section
-        full_segment = afwImage.ImageF(amp_props.full_segment)
+        amp_info = self.camera_info.get_amp_info(amp_name)
+        bbox = self.camera_info.mosaic_section(amp_info)
+        full_segment = afwImage.ImageF(amp_info.getRawBBox())
 
         # Get the imaging segment (i.e., excluding prescan and
         # overscan regions), and fill with data from the eimage.
-        imaging_segment = full_segment.Factory(full_segment, amp_props.imaging)
+        imaging_segment \
+            = full_segment.Factory(full_segment, amp_info.getRawDataBBox())
         data = self.eimage_data[bbox.getMinY():bbox.getMaxY()+1,
                                 bbox.getMinX():bbox.getMaxX()+1].copy()
 
         # Apply flips in x and y relative to assembled eimage in order
         # to have the pixels in readout order.
-        if amp_props.flip_x:
+        if amp_info.getRawFlipX():
             data = data[:, ::-1]
-        if amp_props.flip_y:
+        if amp_info.getRawFlipY():
             data = data[::-1, :]
 
         imaging_segment.getArray()[:] = data
         full_arr = full_segment.getArray()
 
         # Add dark current.
-        full_arr += np.random.poisson(amp_props.dark_current*self.exptime,
+        dark_current = config['electronics_readout']['dark_current']
+        full_arr += np.random.poisson(dark_current*self.exptime,
                                       size=full_arr.shape)
 
         # Add defects.
 
         # Apply CTE.
```

Comment:
I think a bit more comments/documentation here would be a good idea pointing out that the whole matrix is calculated and then applied column-by-column.  It is not super obvious with this compact code.

It could point out how/why both the efficiency and inefficient are used.